gnucash/gnome/dialog-customer.c

```diff
@@ -399,28 +399,22 @@ void
 gnc_customer_name_changed_cb (GtkWidget *widget, gpointer data)
 {
     CustomerWindow *cw = data;
-    char *fullname, *title;
-    const char *id,  *name;
 
     if (!cw)
         return;
 
-    name = gtk_entry_get_text (GTK_ENTRY (cw->company_entry));
+    const char *header = (cw->dialog_type == EDIT_CUSTOMER) ?
+        _("Edit Customer") : _("New Customer");
+
+    const char *name = gtk_entry_get_text (GTK_ENTRY (cw->company_entry));
     if (!name || *name == '\0')
         name = _("<No name>");
 
-    id = gtk_entry_get_text (GTK_ENTRY (cw->id_entry));
-
-    fullname = g_strconcat (name, " (", id, ")", (char *)NULL);
-
-    if (cw->dialog_type == EDIT_CUSTOMER)
-        title = g_strconcat (_("Edit Customer"), " - ", fullname, (char *)NULL);
-    else
-        title = g_strconcat (_("New Customer"), " - ", fullname, (char *)NULL);
+    char *title = g_strdup_printf ("%s - %s (%s)", header, name,
```

Comment:
`g_strdup_printf` is implemented with `g_vasprintf` which in turn calls either the system `vasprintf` if it's available or `vsprintf` otherwise. Being system functions they can't be trusted not to crash if fed a NULL argument, so `fullname` needs to hang around and get tested before being passed to `g_strdup_printf`.